core/controllers/admin.py

```diff
@@ -70,6 +71,16 @@ def get(self):
         self.render_template('admin-page.mainpage.html')
 
 
+class AdminSessionHandler(webapp2.RequestHandler):
+    """Handler for creating and registering the user"""
+
+    def get(self, request, response):
+        auth_services.establish_auth_session(self.request, self.response)
+        auth_claims = auth_services.get_auth_claims_from_request(request)
+        user_id = user_services.create_new_user(
+            auth_claims.auth_id, auth_claims.email)
```

Comment:
establish_auth_session assigns HTTP cookies onto the response, so try to redirect so that the cookies stick around. If this causes an infinite loop, we'll have to look for another solution.
```suggestion
        claims = auth_services.get_auth_claims_from_request(request)
        if claims is None:
            auth_services.establish_auth_session(self.request, self.response)
            self.redirect(self.request.uri)
        else:
            user_id = user_services.create_new_user(
                auth_claims.auth_id, auth_claims.email)
```

Reply:
Done